src/components/chat.tsx

```diff
@@ -0,0 +1,53 @@
+// @ts-nocheck
+// import type { NextPage } from 'next';
+// import { PropsWithChildren } from 'react';
+import Image from 'next/image';
+import avatarReceiver from '../images/avatar/male.png';
+import avatarSender from '../images/avatar/female.png';
+
+// NextJs requires more explicit typing for props when defining a `NextPage`
+const Chat = (props) => {
```

Comment:
I'm not really sure how to word this but some parts are basically the same in this component, which means it can be abstracted even further. For example, line 14 and line 33 are the exact same thing in this component, which means that we can put this outside the logic of the sender vs receiver. The only distinct change I see here is the `justify-end` in line 34. You can basically add `className={receiver === 'receiver' ? "flex items-end justify-end" : "flex items-end"}`.